### PR TITLE
Fix JSON ButtonEvent

### PR DESCRIPTION
**Pull request by a1rwulf**
# RFC
This is an RFC on how to fix the issue.
I see multiple ways to do it, for example:
The CKey constructor could use holdtime and check if it exceeds
the treshold and apply the modifier by itself.
I'm not sure where the longpress translation belongs to from an
architectural pov.

## Description
The current ButtonEvent implementation has 2 problems:
* If any key has a longpress mapping it isn't handled at all
* No matter what holdtime you set, it's not respected

## Types of change

- [x] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [ ] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

**Comment by a1rwulf** [comment]
@garbear What's your thoughts on this?

**Comment by a1rwulf** [comment]
Thx for your comments @garbear I'll check out about a new approach.

**Comment by a1rwulf** [comment]
@garbear that's the new version.
I tried to check how to handle it and ended up adding a new class that handles longpresses for the XBMC_BUTTON event.
I moved the TRESHOLD constant to InputTypes so that it can be used for XBMC_BUTTONS and for XBMC_KEYDOWN/XBMC_KEYUP events.

File: xbmc/input/ButtonStat.cpp [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+CButtonStat::CButtonStat() = default;
+CButtonStat::~CButtonStat() = default;
+
+CKey CButtonStat::TranslateKey(CKey key) const
+{
+  uint32_t buttonCode = key.GetButtonCode();
+  if (key.GetHeld() > HOLD_TRESHOLD)
+    buttonCode |= CKey::MODIFIER_LONG;
+
+  CKey translatedKey(buttonCode, key.GetHeld());
```
Comment (eigendude) [inline xbmc/input/ButtonStat.cpp:24]: Can we not do `key.SetButtonCode();`? Is it ok to drop non-code or held CKey properties?

Comment (eigendude) [reply xbmc/input/ButtonStat.cpp:24]: Nevermind, this class is called `CButtonState`, and buttons are scoped to only have a code and held duration.

**Review by eigendude — APPROVED** [review]
Approval, with one request. Can you move CButtonState into a subdirectory of input/?

**Review by garbear — APPROVED** [review]
Approval, that was my work account

**Comment by a1rwulf** [comment]
Alright I've moved the ButtonStat file into a new subdirectory inside input.
Gave it a short runtime test, seems to work as I expect, if I play a movie:

Shows OSD:
```
curl --data-binary '{"jsonrpc": "2.0", "method": "Input.ButtonEvent", "id": 1, "params": { "button": "enter", "keymap": "KB"}}' -H 'content-type: application/json;' http://localhost:8080/jsonrpc
```

Pauses playback:
```
curl --data-binary '{"jsonrpc": "2.0", "method": "Input.ButtonEvent", "id": 1, "params": { "button": "enter", "keymap": "KB", "holdtime": 500}}' -H 'content-type: application/json;' http://localhost:8080/jsonrpc
```

If jenkins is happy, I think it's good to go.

**Pull request merged**